frontend/src/app/modules/challenges/online-editor/online-editor-page/online-editor-page.component.ts

```diff
@@ -52,38 +57,24 @@ export class OnlineEditorPageComponent implements OnDestroy, OnInit {
     }
 
     ngOnInit() {
-        const challengeId = this.activatedRoute.snapshot.params['id'];
-
         this.splitDirection = 'horizontal';
+        this.activatedRoute.paramMap.subscribe((params: ParamMap) => {
+            const challengeId = +params.get('id')!;
 
-        this.challengeService.getChallengeById(challengeId).subscribe({
-            next: (challenge) => {
-                this.languages = challenge.versions.map((v) => v.language.name);
-                this.challenge = challenge;
-                [this.selectedLanguage] = this.languages;
-                this.editorOptions = {
-                    theme: 'custom-theme',
-                    language: this.mapLanguageName(this.selectedLanguage),
-                    minimap: { enabled: false },
-                    automaticLayout: true,
-                    useShadows: false,
-                    wordWrap: 'on',
-                };
-                this.initialSolution = this.getInitialSolutionByLanguage(this.selectedLanguage);
-            },
-            error: () => {
-                this.router.navigateByUrl('/not-found');
-            },
+            this.loadChallenge(challengeId);
         });
     }
 
     onSelectedLanguageChanged($event: string | string[]): void {
         const selectedLang = this.mapLanguageName($event as string);
 
-        this.initialSolution = this.getInitialSolutionByLanguage($event as string);
+        this.initialSolution = this.getInitialSolutionByLanguage($event as string)!;
 
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         (<any>window).monaco.editor.setModelLanguage((<any>window).monaco.editor.getModels()[0], selectedLang);
```

Comment:
and please remove these options here as well

Reply:
and here you mean remove the line or !

Reply:
I mean line 73-74